ui/src/components/common/utils.ts

```diff
@@ -0,0 +1,5 @@
+const vowels = ['a', 'e', 'i', 'o', 'u'];
+
+export const indefiniteArticle = (word: string): string => {
+    return vowels.includes(word[0].toLowerCase()) ? `an ${word}` : `a ${word}`;
```

Comment:
Not sure how much it matters with this small a list, but using a `Set` here (with `Set.has()`) might be faster than `Array.includes()`.